[RHACS][Docs][4.6 and later versions] ROX-25681: Adding Microsoft Sentinel integration documentation (#83766)
openshift/openshift-docs: modules/viewing-the-log-analytics-to-detect-threats.adoc

// Module included in the following assemblies:
//
// * integration/integrating-with-microsoft-sentinel-notifier.adoc

:_mod-docs-content-type: PROCEDURE
[id="viewing-the-log-analytics-to-detect-threats_{context}"]
= Viewing the log analytics to detect threats

By creating a Microsoft sentinel integration, you can view the log analytics to detect threats.

.Prerequisites

> * You have created a data collection rule, log analytics workspace, service principal and sentinel on Microsoft Azure.

Reviewer: nit: Sentinel does not need to be created.

```suggestion
* You have created a data collection rule, log analytics workspace and a service principal on Microsoft Azure.
```

> * You have configured the notifier supports secret or client certificate authentication.

Reviewer: ```suggestion
* You have either a client secret or a client certificate  configured at the service principal for authentication.
```

> * You have created a schema.

Reviewer: ```suggestion
* You have created a log analytics schema with two fields following format. We recommend separate log analytics tables for Audit Logs and Alerts, both data sources use the same schema.

   * `TimeGenerated` - Datetime
   * `msg` - Dynamic

. Example JSON: 
* ```
"TimeGenerated": "2024-09-03T10:56:58.5010069Z",
"msg": {
     "id": "uuid string"
}
```

Author: @SimonBaeumer I think we do this step by uploading the schema to Microsoft Sentinel which is in the Azure portal. If so, we do not need to document this step in our docs. We could just a cross-ref to Azure docs.
Is there a cross-ref link that you could help me with?

Reviewer: It is true that the user needs to upload the schema in Azure, but we define it. Without us documenting the schema the user does not know what to upload to Azure.

.Procedure

. In the {product-title-short} portal, click *Platform Configuration* -> *Integrations*.
. Scroll down to the *Notifier Integrations* section, and then click *Microsoft Sentinel*.
. To create a new integration, click *New integration*.
. In the *Create integration* page, provide the following information:
** *Integration name*: Specify a name for the integration.
** *Log ingestion endpoint*: Enter the data collection endpoint. You can find the endpoint in the Microsoft Azure portal.
+

> For more information, see link:https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/data-collection-rule-overview[Data collection rules (DCRs) in Azure Monitor] (Microsoft Azure documentation).

Reviewer: Great that you linked azure docs!

** *Directory tenant ID*: Enter the tenant ID which uniquely identifies your Azure Active Directory (AAD) within the Microsoft cloud infrastructure. You can find the tenant ID in the Microsoft Azure portal.
+
For more information, see https://learn.microsoft.com/en-us/azure/active-directory-b2c/tenant-management-read-tenant-name[Find tenant name and tenant ID in Azure Active Directory B2C] (Microsoft Azure documentation).

> ** *Application client ID*: Enter the client ID which uniquely identifies the specific application registered within your AAD that needs access to resources. You can find the client ID in the Microsoft Azure portal.

Reviewer: ```suggestion
** *Application client ID*: Enter the client ID which uniquely identifies the specific application registered within your AAD that needs access to resources. You can find the client ID in the Microsoft Entra portal at the Service Principal you created.
```

+
For more information, see link:https://learn.microsoft.com/en-us/security/zero-trust/develop/app-registration[Register applications] (Microsoft Azure documentation).
** Choose the appropriate authentication method:

> *** If you want to use a secret, enter the secret name. You can find the secret in the Microsoft Azure portal.

Reviewer: ```suggestion
*** If you want to use a secret, enter the secret value. You can find the secret in the Microsoft Azure portal.
```

> *** If you want to use a client certificate, enter the certificate ID and private key. You can find the certificate ID and private key in the Microsoft Azure portal.

Reviewer: ```suggestion
*** If you want to use a client certificate, enter the client certificate and private key. You can find the certificate ID and private key in the Microsoft Azure portal.
```

** Optional: Choose the appropriate method to configure the data collection rule configuration:
*** Select the *Enable alert DCR* checkbox, if you want to enable the alert data collection rule configuration. 
+
To create an alert data collection rule, enter the alert data collection rule stream name and ID. You can find the stream name and ID in the Microsoft Azure portal.
*** Select the *Enable audit log DCR* checkbox, if you want to enable audit data collection rule configuration. 
+
To create an audit data collection rule, enter the stream name and ID. You can find the stream name and ID in the Microsoft Azure portal. 
+
For more information, see link:https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/data-collection-rule-overview[Data collection rules (DCRs) in Azure Monitor] (Microsoft Azure documentation).

. Optional: To test the new integration, click *Test*.
. To save the new integration, click *Save*.

.Verification

. In the {product-title-short} portal, click *Platform Configuration* -> *Integrations*.
. Scroll down to the *Notifier Integrations* section,  and then click *Microsoft Sentinel*.

> . In the *Integrations Microsoft Sentinel* page, verify that the new integration has been created.

Reviewer: ```suggestion
. In the *Integrations Microsoft Sentinel* page, verify that the new integration has been created.
. Check that messages receive the correct log tables in your Log Analytics Workspace.
```

// Module included in the following assemblies:
//
// * integration/integrating-with-microsoft-sentinel-notifier.adoc

:_mod-docs-content-type: PROCEDURE
[id="viewing-the-log-analytics-to-detect-threats_{context}"]
= Viewing the log analytics to detect threats

By creating a Microsoft sentinel integration, you can view the log analytics to detect threats.

.Prerequisites

* You have created a data collection rule, log analytics workspace, and service principal on Microsoft Azure.
* You have configured a client secret or client certificate at the service principal for authentication.
* You have created a log analytics schema. 
** To create a schema, upload the following content to Microsoft sentinel: 
+
.Example JSON
[source,json]
----
{
 "TimeGenerated": "2024-09-03T10:56:58.5010069Z", # <1>
 "msg": {  # <2>

>    "id": "1abe30d1-fa3a-xxxx-xxxx-781f0a12228a" # <3>

Reviewer: I've tried to articulate that the `msg` contains a JSON object of the alert, but only added the ID field.

Here is an example of the complete JSON send to sentinel: https://github.com/stackrox/stackrox/blob/master/central/notifiers/microsoftsentinel/data/example-alert.json

Imho we do not need to add the whole JSON here, it would be too costly to document it here. Do you have an idea how to represent it better?
